src/legacy/core_plugins/kibana/public/discover/plugin.ts

```diff
@@ -77,6 +78,7 @@ const embeddableAngularName = 'app/discoverEmbeddable';
  * Discover provides embeddables, those contain a slimmer Angular
  */
 export class DiscoverPlugin implements Plugin<void, void> {
+  private history = createHashHistory();
```

Comment:
Creating hash history on plugin will add a trailing `#` in the url of any other plugin when discover is moved into the NP (faced the same in the Dashboard), so it would be great if you find a way to initialize it only on plugin mount

Reply:
Changed it: https://github.com/elastic/kibana/pull/62761/commits/b226e2597c93561cdbf7e78ec3912ed50440c30a should address it and create history only when it actually used for the first time